sklearn/linear_model/tests/test_huber.py

```diff
@@ -0,0 +1,180 @@
+    y_new = np.concatenate((y, [y[1]], [y[1]], [y[3]]))
+    huber.fit(X_new, y_new)
+    huber_coef = huber.coef_
+    huber_intercept = huber.intercept_
+    huber.fit(X, y, sample_weight=[1, 3, 1, 2, 1])
+    assert_array_almost_equal(huber.coef_, huber_coef, 3)
+    assert_array_almost_equal(huber.intercept_, huber_intercept, 3)
+
+    # Test sparse implementation with sparse weights.
+    # Checking sparse=non_sparse should be covered in the common tests.
+    X_csr = sparse.csr_matrix(X)
+    huber_sparse = HuberRegressor(fit_intercept=True, alpha=0.1)
+    huber_sparse.fit(X_csr, y, sample_weight=[1, 3, 1, 2, 1])
+    assert_array_almost_equal(huber_sparse.coef_, huber_coef, 3)
+
+
+def return_number_outliers(X, y, coef, intercept, scale, epsilon):
+    """Return the number of outliers."""
+    outliers = np.abs(np.dot(X, coef) + intercept - y) > epsilon * scale
+    return np.sum(outliers)
+
+
+def test_huber_scaling_invariant():
+    """Test that outliers filtering is scaling independent."""
+    X, y = make_regression_with_outliers()
+    huber = HuberRegressor(fit_intercept=False, alpha=0.0, n_iter=100,
+                           epsilon=1.35)
+    huber.fit(X, y)
+    n_outliers1 = return_number_outliers(
+        X, y, huber.coef_, huber.intercept_, huber.scale_, huber.epsilon)
+
+    huber.fit(X, 2. * y)
+    n_outliers2 = return_number_outliers(
+        X, 2. * y, huber.coef_, huber.intercept_, huber.scale_, huber.epsilon)
+
+    huber.fit(2. * X, 2. * y)
+    n_outliers3 = return_number_outliers(
+        2. * X, 2. * y, huber.coef_, huber.intercept_, huber.scale_,
+        huber.epsilon)
+
+    random_features = rng.randint(0, 20, (5,))
+    noise = rng.normal(0, 1, 5)
+    X_new = np.copy(X)
+    X_new[:, random_features] /= noise
+    huber.fit(X_new, y)
+    n_outliers4 = return_number_outliers(
+        X_new, y, huber.coef_, huber.intercept_, huber.scale_, huber.epsilon)
+
+    assert_equal(n_outliers1, n_outliers2)
+    assert_equal(n_outliers3, n_outliers4)
+    assert_equal(n_outliers1, n_outliers4)
+
+
+def test_huber_and_sgd_same_results():
+    """Test they should converge to same coefficients for same parameters"""
+
+    X, y = make_regression_with_outliers(n_samples=5, n_features=1)
+
+    # Fit once to find out the scale parameter. Scale down X and y by scale
+    # so that the scale parameter is optimized to 1.0
+    huber = HuberRegressor(fit_intercept=False, alpha=0.0, n_iter=100,
+                           epsilon=1.35)
+    huber.fit(X, y)
+    X_scale = X / huber.scale_
+    y_scale = y / huber.scale_
+    huber.fit(X_scale, y_scale)
+    assert_almost_equal(huber.scale_, 1.0, 3)
+
+    sgdreg = SGDRegressor(
+        alpha=0.0, loss="huber", shuffle=True, random_state=0, n_iter=1000000,
+        fit_intercept=False, epsilon=1.35)
+    sgdreg.fit(X_scale, y_scale)
+    assert_array_almost_equal(huber.coef_, sgdreg.coef_, 2)
+
+
+def test_huber_warm_start():
```

Comment:
I don't undestand this test. The first fit does 10 iterations, the second 5, right?
Can you add a comment explaining the test?